### PR TITLE
Add Unit Test Coverage for ConcurrentMap and Transport Components

### PR DESCRIPTION
**Pull request by karthik120710**
## Description: 
This PR adds comprehensive unit test coverage for critical backend utilities that previously lacked tests:

pkg/util/concurrent-map.go: Added 9 test functions covering Set, Get, Remove, Len, Iterator operations, including concurrent access testing with 100 goroutines to verify thread-safety, and edge cases like overwrites and empty maps.
pkg/transport/transport.go: Added 7 test functions covering Wrapee helper types (NewWrapee, GetObject, GetID) for various resource types including namespaced resources, core group resources, and cluster-scoped resources.

#3950

### CODE REVIEW

**Comment by karthik120710** [comment]
Hi @MikeSpreitzer and @waltforme, could you please take a look at this PR when you have a chance? Thank you!

**Comment by karthik120710** [comment]
Hi @MikeSpreitzer and @waltforme, could you please take a look at this PR , its ready to merge.

File: pkg/util/concurrent-map_test.go [hunk 1]
```diff
@@ -0,0 +1,234 @@
+/*
+Copyright 2024 The KubeStellar Authors.
```
Comment (MikeSpreitzer) [inline pkg/util/concurrent-map_test.go:2]: This file was created in 2026, right?

File: pkg/util/concurrent-map_test.go [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+			return &testError{msg: expectedErr}
+		}
+		return nil
+	})
+
+	if err == nil {
+		t.Errorf("expected error from iterator")
+	}
+
+	if err.Error() != expectedErr {
```
Comment (MikeSpreitzer) [inline pkg/util/concurrent-map_test.go:146]: This fails to test whether `err` is `==` the `error` value returned by the func argument to `Iterator`.

File: pkg/util/concurrent-map_test.go
Comment (MikeSpreitzer) [inline pkg/util/concurrent-map_test.go:1]: I see a lot of individual tests here that could be chained together in one, much shorter, Test function.

File: pkg/transport/transport_test.go [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+package transport
+
+import (
+	"testing"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+func TestNewWrapee(t *testing.T) {
```
Comment (MikeSpreitzer) [inline pkg/transport/transport_test.go:26]: Why does this one lack a more specific name?

File: pkg/transport/transport_test.go [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+			"metadata": map[string]interface{}{
+				"name": "test-config",
+			},
+		},
+	}
+
+	wr1 := Wrapee{Object: obj, CreateOnly: true}
+	if !wr1.CreateOnly {
+		t.Errorf("expected CreateOnly to be true")
+	}
```
Comment (MikeSpreitzer) [inline pkg/transport/transport_test.go:204]: Isn't this covered by `TestNewWrapee`?

File: pkg/transport/transport_test.go [hunk 5]
```diff
@@ -0,0 +1,210 @@
...
+
+	wr2 := Wrapee{Object: obj, CreateOnly: false}
+	if wr2.CreateOnly {
+		t.Errorf("expected CreateOnly to be false")
+	}
```
Comment (MikeSpreitzer) [inline pkg/transport/transport_test.go:209]: Isn't this covered by `TestNewWrapee_CreateOnlyFalse` ?

File: pkg/transport/transport_test.go
Comment (MikeSpreitzer) [inline pkg/transport/transport_test.go:1]: This is much more source code than is needed. One TestXXX function can test multiple conditions, as is already exhibited. One TestXXX function can cover a collection of test cases, varying the api group, the create-only value, and the namespace.

File: pkg/util/concurrent-map_test.go [hunk 6]
```diff
@@ -0,0 +1,234 @@
...
+		go func(i int) {
+			defer wg.Done()
+			cm.Remove(i)
+		}(i)
+	}
+
+	wg.Wait()
+
+	// Verify final state
+	if cm.Len() != 50 {
```
Comment (MikeSpreitzer) [inline pkg/util/concurrent-map_test.go:185]: The condition tested here is not guaranteed to be true. There is no guarantee that the `cm.Remove` calls all happen after the corresponding `cm.Set`.

**Comment by MikeSpreitzer** [comment]
Why did you add this when #3856 already exists?

File: pkg/transport/transport_test.go
Comment (MikeSpreitzer) [inline pkg/transport/transport_test.go:1]: Why did you add this when #3874 already exists?
